### PR TITLE
Don't show "can't redact" dialog on network error, with redaction having local echo & queuing now.

### PR DESCRIPTION
**Pull request by bwindels**
Goes with https://github.com/matrix-org/matrix-js-sdk/pull/937

### CODE REVIEW

**Comment by bwindels** [comment]
still trying to figure out if I can detect a network error when redacting, as ideally we shouldn't show the "cant redact" dialog as the redaction might go out later.

**Comment by bwindels** [comment]
ready for review now.

File: src/components/views/context_menus/MessageContextMenu.js [hunk 1]
```diff
@@ -119,19 +119,29 @@ module.exports = React.createClass({
...
+                    );
+                } catch (e) {
                     const code = e.errcode || e.statusCode;
-                    Modal.createTrackedDialog('You cannot delete this message', '', ErrorDialog, {
-                        title: _t('Error'),
-                        description: _t('You cannot delete this message. (%(code)s)', {code}),
-                    });
-                }).done();
+                    // only show the dialog if failing for something other than a network error
+                    // (e.g. no errcode or statusCode) as that case the redaction end up in the
```
Comment (jryans) [inline src/components/views/context_menus/MessageContextMenu.js]:
```suggestion
                    // (e.g. no errcode or statusCode) as in that case the redactions end up in the
```

**Comment by bwindels** [comment]
Waiting for js-sdk PR to merge.

**Comment by bwindels** [comment]
> This is fine by itself, but did you want to also listen to the new JS SDK event here or in a separate change? (Or is not needed for React SDK at all...?)

Yes, just noticed my push didn't go through because of the suggestion I merged here :smile: Should be good now.

**Pull request merged**